File: frontend/src/types/inventory.ts
import type { ScanError } from './scanner'

export type InventoryItem = {
  id: string
  relativePath: string
  absolutePath: string
  directory: string
  filename: string
  extension: string
  category: string
  size: number
  createdAt: string | null
  modifiedAt: string | null
  isDirectory: false
  readable: boolean
}

export type InventoryFolder = { path: string; relativePath: string; name: string; parent: string | null; depth: number }
export type InventoryStatistics = { totalFiles: number; totalDirectories: number; totalSize: number; totalImages: number; totalPages: number; totalVideos: number; totalDocuments: number; totalArchives: number; totalData: number; totalOther: number }
export type InventoryCategorySummary = { category: string; count: number }
export type InventoryExtensionSummary = { extension: string; count: number }
export type InventoryMetadata = { generatedAt: string; scannerVersion: string; workspacePath: string; durationMs: number }
export type Inventory = { metadata: InventoryMetadata; statistics: InventoryStatistics; folders: InventoryFolder[]; items: InventoryItem[]; categories: InventoryCategorySummary[]; extensions: InventoryExtensionSummary[]; errors: ScanError[] }
export type InventoryRequest = { workspacePath: string }

File: football-collection-builder/backend/app/models/inventory.py
from pydantic import BaseModel

from app.models.scanner import ScanError


class InventoryItem(BaseModel):
    id: str
    relativePath: str
    absolutePath: str
    directory: str
    filename: str
    extension: str
    category: str
    size: int
    createdAt: str | None = None
    modifiedAt: str | None = None
    isDirectory: bool = False
    readable: bool


class InventoryFolder(BaseModel):
    path: str
    relativePath: str
    name: str
    parent: str | None = None
    depth: int


class InventoryStatistics(BaseModel):
    totalFiles: int
    totalDirectories: int
    totalSize: int
    totalImages: int
    totalPages: int
    totalVideos: int
    totalDocuments: int
    totalArchives: int
    totalData: int
    totalOther: int


class InventoryCategorySummary(BaseModel):
    category: str
    count: int


class InventoryExtensionSummary(BaseModel):
    extension: str
    count: int


class InventoryMetadata(BaseModel):
    generatedAt: str
    scannerVersion: str
    workspacePath: str
    durationMs: int


class Inventory(BaseModel):
    metadata: InventoryMetadata
    statistics: InventoryStatistics
    folders: list[InventoryFolder]
    items: list[InventoryItem]
    categories: list[InventoryCategorySummary]
    extensions: list[InventoryExtensionSummary]
    errors: list[ScanError]


class InventoryRequest(BaseModel):
    workspacePath: str

class InventoryPersistenceResult(BaseModel):
    success: bool
    databaseCreated: bool
    savedAt: str
    fileCount: int
    folderCount: int
    durationMs: int
    message: str


class InventoryDatabaseStatus(BaseModel):
    databaseCreated: bool
    lastSavedAt: str | None = None
    fileCount: int = 0
    folderCount: int = 0